src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -185,21 +188,58 @@ public TripMonitorNotification checkTripForNewAlerts() {
         return notification;
     }
 
-    public TripMonitorNotification checkTripForDepartureDelay() {
-        long departureDelayInMinutes = TimeUnit.SECONDS.toMinutes(matchingItinerary.legs.get(0).departureDelay);
-        // First leg departure time should not exceed variance allowed.
-        if (departureDelayInMinutes >= trip.departureVarianceMinutesThreshold) {
-            return TripMonitorNotification.createDelayNotification(departureDelayInMinutes, trip.departureVarianceMinutesThreshold, DEPARTURE_DELAY);
+    /**
+     * Checks whether the trip is beginning or ending at a time greater than the allowable variance. This will check
+     * whether the departure or arrival time of the whole journey has deviated to the point where the absolute value of
+     * the variance has changed more than the trip's variance. If it has, a notification is generated.
+     */
+    public TripMonitorNotification checkTripForDelay(NotificationType delayType) {
+        // the target time for trip depending on notification type
+        Date matchingItineraryTargetTime;
+
+        // the current baseline epoch millis to check if a new threshold has been met
+        long baselineItineraryTargetEpochMillis;
```

Comment:
"baseline" feels too close to "original" to me. Maybe "latest"?

Reply:
`latest` doesn't make sense to me, because this isn't always the most recently obtained time. `baseline` on the other has a definition of "a specific value or values that can serve as a comparison or control" which seems very appropriate here.

Reply:
There's still not a great definition of/distinction between original vs. baseline in the javadoc. If we're keeping those variable names, I think they need better descriptions both here and in journey state fields. Maybe it makes perfect sense to use these names, but I'm still finding this somewhat hard to follow.